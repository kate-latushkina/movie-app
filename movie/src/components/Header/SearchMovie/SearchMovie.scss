.search-panel {
  &_input {
    padding: 0 20px;
    &::placeholder {
      font-size: 17px;
      padding: 5px 0;
    }
  }

  &_button {
    width: 20%;
    height: 100%;
  }
}
