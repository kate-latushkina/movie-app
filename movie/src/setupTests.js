// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Enzyme, { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.render = render;
global.mount = mount;
global.toJson = toJson;

// Fail tests on any warning
console.error = message => {
  throw new Error(message);
};
