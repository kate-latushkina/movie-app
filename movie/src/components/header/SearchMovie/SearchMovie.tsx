import { Button, Grid } from "@material-ui/core";
import React from "react";
import styled from "styled-components";
import './SearchMovie.scss'

const SearchMovie: React.FC = () => {
  return (
    <SearchPanel container justify="space-between">
      <Input className="search-panel_input" placeholder="What do you want to watch?"/>
      <Button variant="contained" color="secondary" className="search-panel_button">Search</Button>
    </SearchPanel>
  );
};

const SearchPanel = styled(Grid)`
  width: 100%;
  height: 40px;
`;

const Input = styled.input`
  width 70%;
  height: 100%;
  border-radius: 2px;
  border: none;
`;


export default SearchMovie;
