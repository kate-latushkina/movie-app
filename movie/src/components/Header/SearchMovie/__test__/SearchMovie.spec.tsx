import React from "react";
import { shallow } from "enzyme";
import SearchMovie from "../SearchMovie";
import "../../../../setupTests";

describe("should render SearchMovie component", () => {
  let component: any;

  beforeEach(() => {
    component = shallow(<SearchMovie />);
  });

  it("should contain Button wrapper", () => {
    const wrapper = component.find(".search-panel_button");
    expect(wrapper.length).toBe(1);
  });
  it("should contain Input wrapper", () => {
    const wrapper = component.find(".search-panel_input");
    expect(wrapper.length).toBe(1);
  });
});
