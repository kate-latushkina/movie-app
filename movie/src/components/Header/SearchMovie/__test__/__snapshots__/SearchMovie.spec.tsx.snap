// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchMovie component should render SearchMovie snapshot 1`] = `
<div
  className="MuiGrid-root sc-bdfBwQ nKNBL MuiGrid-container MuiGrid-justify-xs-space-between"
>
  <input
    className="sc-gsTCUz gbGCtm search-panel_input"
    placeholder="What do you want to watch?"
  />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained search-panel_button MuiButton-containedSecondary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Search
    </span>
  </button>
</div>
`;
